baremetal_network_provisioning/drivers/snmp_client.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2015 OpenStack Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from baremetal_network_provisioning.common import constants
+
+from ironic.common import exception
```

Comment:
why ironic.common ?
it should be removed . move SNMPFailure exception to the baremetal repo common/exceptions.py